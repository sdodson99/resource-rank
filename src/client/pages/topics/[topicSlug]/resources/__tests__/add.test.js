import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import getTopicBySlug from '@/services/topics/graphql-topic-by-slug-service';
import { when } from 'jest-when';
import AddTopicResource, { Page, getServerSideProps } from '../add.page';
import useNavigate from '@/hooks/use-navigate';
import useAvailableTopicResourceSearch from '@/hooks/topics/use-available-topic-resource-search';
import useTopicResourceCreator from '@/hooks/topics/use-topic-resource-creator';
import withApp from '@/test-utils/with-app';

jest.mock('@/services/topics/graphql-topic-by-slug-service');
jest.mock('@/hooks/use-navigate');
jest.mock('@/hooks/topics/use-available-topic-resource-search');
jest.mock('@/hooks/topics/use-topic-resource-creator');

describe('<AddTopicResource />', () => {
  describe('page', () => {
    let topicId;
    let topicName;
    let topicSlug;
    let props;

    let mockAvailableTopicResourceSearch;
    let mockNavigate;

    beforeEach(() => {
      mockAvailableTopicResourceSearch = {
        data: {
          availableResources: {
            items: [
              {
                resource: {
                  id: '1',
                  slug: 'slug-1',
                },
                alreadyAdded: false,
              },
              {
                resource: {
                  id: '2',
                  slug: 'slug-2',
                },
                alreadyAdded: true,
              },
            ],
          },
        },
        error: null,
        isLoading: false,
        isInitialized: true,
        searchVariables: { search: 'search' },
        currentSearchVariables: { search: null, limit: 10 },
        debounceProcessSearch: jest.fn(),
        currentPage: 1,
        processPageNumber: jest.fn(),
      };
      useAvailableTopicResourceSearch.mockReturnValue(
        mockAvailableTopicResourceSearch
      );
      useTopicResourceCreator.mockReturnValue({});

      mockNavigate = jest.fn();
      useNavigate.mockReturnValue(mockNavigate);

      topicId = '123';
      topicName = 'topic-name';
      topicSlug = 'topic-slug';
      props = {
        topicId,
        topicName,
        topicSlug,
      };
    });

    afterEach(() => {
      useNavigate.mockReset();
      useAvailableTopicResourceSearch.mockReset();
      useTopicResourceCreator.mockReset();
    });

    it('should mount', () => {
      render(withApp(Page, props));

      const page = screen.getByTestId('AddTopicResourcePage');

      expect(page).toBeInTheDocument();
    });

    it('should process search on search input', () => {
      const search = '123';
      render(withApp(Page, props));
      const searchInput = screen.getByTestId('SearchInput');

      fireEvent.input(searchInput, {
        target: {
          value: search,
        },
      });

      expect(
        mockAvailableTopicResourceSearch.debounceProcessSearch
      ).toBeCalledWith({
        search,
        offset: 0,
        limit: 10,
      });
    });

    it('should show resources when available resources data loads', async () => {
      render(withApp(Page, props));

      await waitFor(() => {
        const availableResourceListingItems = screen.getAllByTestId(
          'AvailableResourceListingItem'
        );
        expect(availableResourceListingItems.length).toBe(2);
      });
    });

    it('should show loading when topic resources initializing', () => {
      useAvailableTopicResourceSearch.mockReturnValue({
        ...mockAvailableTopicResourceSearch,
        isInitialized: false,
      });
      render(withApp(Page, props));

      const loadingDisplay = screen.getByTestId('SkeletonListing');

      expect(loadingDisplay).toBeInTheDocument();
    });

    describe('onAddResource', () => {
      function renderAndSubmit() {
        render(withApp(Page, props));

        const addResourceButton = screen.getAllByTestId('AddResourceButton')[0];
        addResourceButton.click();
      }

      let mockCreate;
      let resourceId;

      beforeEach(() => {
        mockCreate = jest.fn();
        useTopicResourceCreator.mockReturnValue({
          createTopicResource: mockCreate,
        });

        resourceId = '1';
      });

      it('should renavigate to new topic resource if successful', async () => {
        when(mockCreate).calledWith(topicId, resourceId).mockReturnValue(true);

        renderAndSubmit();

        await waitFor(() => {
          expect(mockNavigate).toBeCalledWith({
            pathname: `/topics/topic-slug/resources/slug-1`,
            query: {
              new: true,
            },
          });
        });
      });

      it('should display resource error if not successful', async () => {
        when(mockCreate).calledWith(topicId, resourceId).mockReturnValue(false);

        renderAndSubmit();

        await waitFor(() => {
          const errorMessage = screen.getByText(
            'Failed to add topic resource.'
          );
          expect(errorMessage).toBeInTheDocument();
        });
      });

      it('should clear resource error if successful', async () => {
        when(mockCreate).calledWith(topicId, resourceId).mockReturnValue(false);
        renderAndSubmit();
        await waitFor(() => {
          const errorMessage = screen.getByText(
            'Failed to add topic resource.'
          );
          expect(errorMessage).toBeInTheDocument();
        });
        when(mockCreate).calledWith(topicId, resourceId).mockReturnValue(true);

        renderAndSubmit();

        await waitFor(() => {
          const errorMessage = screen.queryByText(
            'Failed to add topic resource.'
          );
          expect(errorMessage).toBeNull();
        });
      });
    });

    it('should render correctly', () => {
      const page = renderer.create(withApp(Page, props)).toJSON();

      expect(page).toMatchSnapshot();
    });

    it('should render correctly with current search', () => {
      mockAvailableTopicResourceSearch.currentSearchVariables = {
        search: 'currentSearch',
      };
      useAvailableTopicResourceSearch.mockReturnValue(
        mockAvailableTopicResourceSearch
      );

      const page = renderer.create(withApp(Page, props)).toJSON();

      expect(page).toMatchSnapshot();
    });
  });

  describe('HOC page', () => {
    it('should require authentication', () => {
      const page = renderer
        .create(withApp(AddTopicResource, {}, { mock: 'unauthenticated' }))
        .toJSON();

      expect(page).toMatchSnapshot();
    });
  });

  describe('getServerSideProps', () => {
    let req;
    let params;
    let query;
    let mock;
    let topicSlug;

    beforeEach(() => {
      mock = 'mock';
      query = {
        mock,
      };
      req = {};
      topicSlug = 'topic-name';
      params = {
        topicSlug,
      };
    });

    afterEach(() => {
      getTopicBySlug.mockReset();
    });

    it('should return not found if topic query fails', async () => {
      when(getTopicBySlug)
        .calledWith(topicSlug, { mock })
        .mockImplementation(() => {
          throw new Error();
        });

      const { notFound } = await getServerSideProps({ req, params, query });

      expect(notFound).toBeTruthy();
    });

    describe('when successful', () => {
      let id;
      let name;
      let slug;

      beforeEach(() => {
        id = '123';
        name = 'name';
        slug = 'slug';
      });

      it('should return topic props', async () => {
        when(getTopicBySlug).calledWith(topicSlug, { mock }).mockReturnValue({
          id,
          name,
          slug,
        });

        const { props } = await getServerSideProps({ req, params, query });

        expect(props).toEqual({
          topicId: id,
          topicName: name,
          topicSlug: slug,
        });
      });
    });
  });
});
