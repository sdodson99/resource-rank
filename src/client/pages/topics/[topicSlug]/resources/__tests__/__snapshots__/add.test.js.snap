// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddTopicResource /> HOC page should require authentication 1`] = `
<div
  className="Layout"
  data-testid="Layout"
>
  <div
    className="Page"
  >
    <header
      className="Header"
      data-testid="Header"
    >
      <div
        className="content-container"
      >
        <div
          className="sm:flex items-center text-center sm:text-left"
        >
          <div
            className="flex-grow flex justify-center sm:justify-start"
          >
            <a
              href="/?mock=unauthenticated"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              <img
                alt="Resource Rank Logo"
                height="50"
                src="/img/logo.svg"
                width="50"
              />
            </a>
          </div>
          <div
            className="NavItem NavLink ActiveNavLink"
            data-testid="ActiveLink"
          >
            Home
          </div>
          <div
            className="NavItem"
            data-testid="TopicsNavItem"
          >
            <div
              className="NavLink"
              data-testid="ActiveLink"
            >
              <a
                className=""
                href="/topics?mock=unauthenticated"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                Topics
              </a>
            </div>
          </div>
          <div
            className="NavItem"
          >
            <button
              className="btn btn-primary"
              data-testid="LoadingButton"
              onClick={[Function]}
            >
              Login
            </button>
          </div>
        </div>
      </div>
    </header>
    <div>
      <div
        className="mt-8"
      >
        <div
          className="LoadingSpinner"
          data-testid="LoadingSpinner"
        >
          <svg
            fill="none"
            height={50}
            viewBox="0 0 100 100"
            width={50}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M95 50C95 59.6293 91.9111 69.0047 86.1874 76.7483C80.4636 84.4919 72.4069 90.1952 63.2012 93.0201C53.9956 95.845 44.1265 95.6424 35.0445 92.4421C25.9625 89.2419 18.1466 83.2127 12.7454 75.2408C7.34427 67.2689 4.64275 57.7746 5.03791 48.1534C5.43306 38.5321 8.90405 29.2914 14.9407 21.7892C20.9774 14.287 29.2615 8.91914 38.5753 6.47442C47.8891 4.0297 57.7416 4.6371 66.6846 8.20736"
              stroke="#292929"
              strokeWidth="10"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
  <footer
    className="Footer"
    data-testid="Footer"
  >
    <div
      className="content-container"
    >
      <div
        className="sm:flex text-center sm:text-left"
      >
        <div
          className="sm:order-2 sm:ml-10"
        >
          <h1
            className="text-4xl"
          >
            Our Mission
          </h1>
          <p
            className="py-3"
          >
            is to 
            <strong>
              simplify
            </strong>
             your learning journey. Learning can be 
            <strong>
              difficult
            </strong>
            , especially when starting. There are
             
            <strong>
              millions
            </strong>
             of educational courses, videos, and books
             
            <strong>
              scattered
            </strong>
             across the Internet.
          </p>
          <p
            className="py-3"
          >
            Resource Rankâ€™s goal is to 
            <strong>
              consolidate
            </strong>
             all of these resources so that you can 
            <strong>
              quickly
            </strong>
             find the
             
            <strong>
              best
            </strong>
             resources for what you are learning.
          </p>
        </div>
        <div
          className="sm:order-1 mt-10 sm:mt-0 flex items-start justify-center sm:justify-start"
        >
          <img
            alt="Resource Rank Logo"
            className="max-w-none"
            height="75"
            src="/img/logo.svg"
            width="75"
          />
        </div>
      </div>
    </div>
  </footer>
</div>
`;

exports[`<AddTopicResource /> page should render correctly 1`] = `
<div
  data-testid="AddTopicResourcePage"
>
  <div
    className="BreadcrumbLayout"
    data-testid="BreadcrumbLayout"
  >
    <div
      className="Layout"
      data-testid="Layout"
    >
      <div
        className="Page"
      >
        <header
          className="Header"
          data-testid="Header"
        >
          <div
            className="content-container"
          >
            <div
              className="sm:flex items-center text-center sm:text-left"
            >
              <div
                className="flex-grow flex justify-center sm:justify-start"
              >
                <a
                  href="/?mock=standard"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  <img
                    alt="Resource Rank Logo"
                    height="50"
                    src="/img/logo.svg"
                    width="50"
                  />
                </a>
              </div>
              <div
                className="NavItem NavLink ActiveNavLink"
                data-testid="ActiveLink"
              >
                Home
              </div>
              <div
                className="NavItem"
                data-testid="TopicsNavItem"
              >
                <div
                  className="NavLink"
                  data-testid="ActiveLink"
                >
                  <a
                    className=""
                    href="/topics?mock=standard"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                  >
                    Topics
                  </a>
                </div>
              </div>
              <div
                className="NavItem"
              >
                <button
                  className="btn btn-primary"
                  data-testid="LoadingButton"
                  onClick={[Function]}
                >
                  Login
                </button>
              </div>
            </div>
          </div>
        </header>
        <div>
          <div
            className="content-container pb-10"
          >
            <nav
              aria-label="breadcrumb"
              className="BreadcrumbContent"
            >
              <ol
                className="flex flex-wrap"
              >
                <li
                  className="flex"
                >
                  <div
                    className="BreadcrumbItem"
                    data-testid="ActiveLink"
                  >
                    <a
                      className="hyperlink"
                      href="/topics?mock=standard"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                    >
                      Topics
                    </a>
                  </div>
                  <div
                    className="px-3"
                  >
                    /
                  </div>
                </li>
                <li
                  className="flex"
                >
                  <div
                    className="BreadcrumbItem"
                    data-testid="ActiveLink"
                  >
                    <a
                      className="hyperlink"
                      href="/topics/topic-slug?mock=standard"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                    >
                      topic-name
                    </a>
                  </div>
                  <div
                    className="px-3"
                  >
                    /
                  </div>
                </li>
                <li
                  className="flex"
                >
                  <div
                    className="BreadcrumbItem"
                    data-testid="ActiveLink"
                  >
                    <a
                      className="hyperlink"
                      href="/topics/topic-slug/resources/add?mock=standard"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                    >
                      Add
                    </a>
                  </div>
                </li>
              </ol>
            </nav>
            <div>
              <div
                className="PageHeaderButton"
                data-testid="PageHeaderButton"
              >
                <div
                  className="flex flex-col items-center justify-between text-center sm:flex-row sm:text-left"
                >
                  <div
                    className="text-4xl"
                  >
                    Add Resource
                  </div>
                  <div
                    className="mt-3 flex sm:mt-0 sm:ml-3"
                  >
                    <a
                      className="btn btn-primary"
                      href="/topics/topic-slug/resources/new?mock=standard"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                    >
                      New
                    </a>
                  </div>
                </div>
              </div>
              <div
                className="mt-8 flex flex-col"
              >
                <input
                  className="flex-grow form-control-lg"
                  data-testid="SearchInput"
                  onChange={[Function]}
                  placeholder="Search resources..."
                  type="text"
                  value="search"
                />
              </div>
              <div
                className="mt-8"
              >
                <div
                  className="LoadingErrorEmptyDataLayout"
                  data-testid="LoadingErrorEmptyDataLayout"
                >
                  <div>
                    <div>
                      <div
                        className="text-center sm:text-left"
                      >
                        No resources are available to add.
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <footer
        className="Footer"
        data-testid="Footer"
      >
        <div
          className="content-container"
        >
          <div
            className="sm:flex text-center sm:text-left"
          >
            <div
              className="sm:order-2 sm:ml-10"
            >
              <h1
                className="text-4xl"
              >
                Our Mission
              </h1>
              <p
                className="py-3"
              >
                is to 
                <strong>
                  simplify
                </strong>
                 your learning journey. Learning can be 
                <strong>
                  difficult
                </strong>
                , especially when starting. There are
                 
                <strong>
                  millions
                </strong>
                 of educational courses, videos, and books
                 
                <strong>
                  scattered
                </strong>
                 across the Internet.
              </p>
              <p
                className="py-3"
              >
                Resource Rankâ€™s goal is to 
                <strong>
                  consolidate
                </strong>
                 all of these resources so that you can 
                <strong>
                  quickly
                </strong>
                 find the
                 
                <strong>
                  best
                </strong>
                 resources for what you are learning.
              </p>
            </div>
            <div
              className="sm:order-1 mt-10 sm:mt-0 flex items-start justify-center sm:justify-start"
            >
              <img
                alt="Resource Rank Logo"
                className="max-w-none"
                height="75"
                src="/img/logo.svg"
                width="75"
              />
            </div>
          </div>
        </div>
      </footer>
    </div>
  </div>
</div>
`;

exports[`<AddTopicResource /> page should render correctly with current search 1`] = `
<div
  data-testid="AddTopicResourcePage"
>
  <div
    className="BreadcrumbLayout"
    data-testid="BreadcrumbLayout"
  >
    <div
      className="Layout"
      data-testid="Layout"
    >
      <div
        className="Page"
      >
        <header
          className="Header"
          data-testid="Header"
        >
          <div
            className="content-container"
          >
            <div
              className="sm:flex items-center text-center sm:text-left"
            >
              <div
                className="flex-grow flex justify-center sm:justify-start"
              >
                <a
                  href="/?mock=standard"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  <img
                    alt="Resource Rank Logo"
                    height="50"
                    src="/img/logo.svg"
                    width="50"
                  />
                </a>
              </div>
              <div
                className="NavItem NavLink ActiveNavLink"
                data-testid="ActiveLink"
              >
                Home
              </div>
              <div
                className="NavItem"
                data-testid="TopicsNavItem"
              >
                <div
                  className="NavLink"
                  data-testid="ActiveLink"
                >
                  <a
                    className=""
                    href="/topics?mock=standard"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                  >
                    Topics
                  </a>
                </div>
              </div>
              <div
                className="NavItem"
              >
                <button
                  className="btn btn-primary"
                  data-testid="LoadingButton"
                  onClick={[Function]}
                >
                  Logout
                </button>
              </div>
            </div>
          </div>
        </header>
        <div>
          <div
            className="content-container pb-10"
          >
            <nav
              aria-label="breadcrumb"
              className="BreadcrumbContent"
            >
              <ol
                className="flex flex-wrap"
              >
                <li
                  className="flex"
                >
                  <div
                    className="BreadcrumbItem"
                    data-testid="ActiveLink"
                  >
                    <a
                      className="hyperlink"
                      href="/topics?mock=standard"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                    >
                      Topics
                    </a>
                  </div>
                  <div
                    className="px-3"
                  >
                    /
                  </div>
                </li>
                <li
                  className="flex"
                >
                  <div
                    className="BreadcrumbItem"
                    data-testid="ActiveLink"
                  >
                    <a
                      className="hyperlink"
                      href="/topics/topic-slug?mock=standard"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                    >
                      topic-name
                    </a>
                  </div>
                  <div
                    className="px-3"
                  >
                    /
                  </div>
                </li>
                <li
                  className="flex"
                >
                  <div
                    className="BreadcrumbItem"
                    data-testid="ActiveLink"
                  >
                    <a
                      className="hyperlink"
                      href="/topics/topic-slug/resources/add?mock=standard"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                    >
                      Add
                    </a>
                  </div>
                </li>
              </ol>
            </nav>
            <div>
              <div
                className="PageHeaderButton"
                data-testid="PageHeaderButton"
              >
                <div
                  className="flex flex-col items-center justify-between text-center sm:flex-row sm:text-left"
                >
                  <div
                    className="text-4xl"
                  >
                    Add Resource
                  </div>
                  <div
                    className="mt-3 flex sm:mt-0 sm:ml-3"
                  >
                    <a
                      className="btn btn-primary"
                      href="/topics/topic-slug/resources/new?mock=standard"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                    >
                      New
                    </a>
                  </div>
                </div>
              </div>
              <div
                className="mt-8 flex flex-col"
              >
                <input
                  className="flex-grow form-control-lg"
                  data-testid="SearchInput"
                  onChange={[Function]}
                  placeholder="Search resources..."
                  type="text"
                  value="search"
                />
              </div>
              <div
                className="mt-8"
              >
                <div
                  className="LoadingErrorEmptyDataLayout"
                  data-testid="LoadingErrorEmptyDataLayout"
                >
                  <div>
                    <div>
                      <div
                        className="AvailableResourceListing"
                        data-testid="AvailableResourceListing"
                      >
                        <div
                          className="PaginatedListing"
                          data-testid="PaginatedListing"
                        >
                          <div
                            className="Listing"
                            data-testid="Listing"
                          >
                            <div
                              className="AvailableResourceListingItem"
                              data-testid="AvailableResourceListingItem"
                            >
                              <div
                                className="ListingItem"
                                data-testid="ListingItem"
                              >
                                <div
                                  className="text-center sm:text-left sm:flex sm:justify-between sm:items-center"
                                >
                                  <div>
                                    <div
                                      className="flex items-center justify-center sm:justify-start"
                                    >
                                      <div
                                        className=""
                                      />
                                    </div>
                                  </div>
                                  <div
                                    className="mt-5 sm:mt-0 sm:ml-5"
                                  >
                                    <button
                                      className="btn btn-primary"
                                      data-testid="AddResourceButton"
                                      onClick={[Function]}
                                    >
                                      <svg
                                        aria-hidden={true}
                                        className="MuiSvgIcon-root"
                                        fill="white"
                                        focusable="false"
                                        viewBox="0 0 24 24"
                                      >
                                        <path
                                          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                                        />
                                      </svg>
                                    </button>
                                  </div>
                                </div>
                              </div>
                            </div>
                            <div
                              className="AvailableResourceListingItem"
                              data-testid="AvailableResourceListingItem"
                            >
                              <div
                                className="ListingItem"
                                data-testid="ListingItem"
                              >
                                <div
                                  className="text-center sm:text-left sm:flex sm:justify-between sm:items-center"
                                >
                                  <div>
                                    <div
                                      className="flex items-center justify-center sm:justify-start"
                                    >
                                      <div
                                        className="line-through"
                                      />
                                    </div>
                                    <div
                                      className="error-text italic text-xs"
                                    >
                                      Already added
                                    </div>
                                  </div>
                                  <div
                                    className="mt-5 sm:mt-0 sm:ml-5"
                                  >
                                    <button
                                      className="btn btn-primary"
                                      data-testid="AddResourceButton"
                                      disabled={true}
                                      onClick={[Function]}
                                    >
                                      <svg
                                        aria-hidden={true}
                                        className="MuiSvgIcon-root"
                                        fill="white"
                                        focusable="false"
                                        viewBox="0 0 24 24"
                                      >
                                        <path
                                          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                                        />
                                      </svg>
                                    </button>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                          <div
                            className="mt-8 flex justify-center"
                          >
                            <div
                              className="Pagination"
                              data-testid="Pagination"
                            >
                              <button
                                className="SelectedPage"
                                data-testid="Page"
                                disabled={true}
                                onClick={[Function]}
                                type="button"
                              >
                                1
                              </button>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <footer
        className="Footer"
        data-testid="Footer"
      >
        <div
          className="content-container"
        >
          <div
            className="sm:flex text-center sm:text-left"
          >
            <div
              className="sm:order-2 sm:ml-10"
            >
              <h1
                className="text-4xl"
              >
                Our Mission
              </h1>
              <p
                className="py-3"
              >
                is to 
                <strong>
                  simplify
                </strong>
                 your learning journey. Learning can be 
                <strong>
                  difficult
                </strong>
                , especially when starting. There are
                 
                <strong>
                  millions
                </strong>
                 of educational courses, videos, and books
                 
                <strong>
                  scattered
                </strong>
                 across the Internet.
              </p>
              <p
                className="py-3"
              >
                Resource Rankâ€™s goal is to 
                <strong>
                  consolidate
                </strong>
                 all of these resources so that you can 
                <strong>
                  quickly
                </strong>
                 find the
                 
                <strong>
                  best
                </strong>
                 resources for what you are learning.
              </p>
            </div>
            <div
              className="sm:order-1 mt-10 sm:mt-0 flex items-start justify-center sm:justify-start"
            >
              <img
                alt="Resource Rank Logo"
                className="max-w-none"
                height="75"
                src="/img/logo.svg"
                width="75"
              />
            </div>
          </div>
        </div>
      </footer>
    </div>
  </div>
</div>
`;
